### PR TITLE
Update to .NET 6

### PR DESCRIPTION
**Pull request by yaqian256**
# Justification
Update analyzers to latest version to better support .NET 6.

# Implementation
- Updated analyzer references to be the latest stable version
- Removed `Microsoft.DotNet.Analyzers.Compatibility` because it's been replaced with the built-in analyzers of .NET 5 and later versions.
- Fix new warnings in the code
- Add `global.json`

# Testing
Ensured code builds and tests pass

### CODE REVIEW

File: src/NationalInstruments.Analyzers.Utilities/Renamer.cs [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-            CancellationToken cancellationToken = default(CancellationToken))
-        {
-            if (document == null)
-            {
-                throw new ArgumentNullException(nameof(document));
-            }
-
-            Solution solution = document.Project.Solution;
-
-            return Microsoft.CodeAnalysis.Rename.Renamer.RenameSymbolAsync(
```
Comment (yaqian256) [inline src/NationalInstruments.Analyzers.Utilities/Renamer.cs:30]: NOTE: this RenameSymbolAsync method (with the `OptionSet` parameter) has been obsoleted and caused build failures. I changed the caller in `FieldsCamelCasedWithUnderscoreCodeFixProvider` to call the new `RenameSymbolAsync` method (with `SymbolRenameOptions`) directly.

Deleted this class because Roslynator had also removed it since v1.3.10.

File: src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs [hunk 2]
```diff
@@ -67,7 +67,7 @@ private void AnalyzeUsingStatement(SyntaxNodeAnalysisContext context)
             var usingStatementSyntax = (UsingStatementSyntax)context.Node;
             var containingMethodSyntax = usingStatementSyntax.Ancestors().OfType<BaseMethodDeclarationSyntax>().FirstOrDefault();
             if (containingMethodSyntax != null &&
-                !containingMethodSyntax.Modifiers.Any(syntaxToken => syntaxToken.Kind() == SyntaxKind.AsyncKeyword))
+                !containingMethodSyntax.Modifiers.Any(syntaxToken => syntaxToken.IsKind(SyntaxKind.AsyncKeyword)))
```
Comment (yaqian256) [inline src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs]: NOTE: this is to fix a warning reported by the new csharp analyzer.

File: Directory.Build.props [hunk 3]
```diff
@@ -58,15 +54,11 @@
...
-        <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" />
-        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.11.0" />
-        <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.2" />
-        <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="16.10.56" />
+        <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0" />
+        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.2.0" />
+        <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3" />
+        <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="17.2.32" />
         <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435" />
-        <PackageReference Condition=" '$(IncludeCompatabilityAnalyzer)' == 'True' " Include="Microsoft.DotNet.Analyzers.Compatibility" Version="0.2.12-alpha">
```
Comment (yaqian256) [inline Directory.Build.props:66]: NOTE: Removed this because it's been replaced with the built-in analyzers of .NET 5 and later versions,  according to https://www.nuget.org/packages/Microsoft.DotNet.Analyzers.Compatibility/0.2.12-alpha

File: Directory.Build.props [hunk 4]
```diff
@@ -58,15 +54,11 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" />
-        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.11.0" />
-        <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.2" />
-        <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="16.10.56" />
+        <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0" />
```
Comment (nick-beer) [inline Directory.Build.props]: I think I'd remove this one - per the recommendation in the note [here](https://docs.microsoft.com/en-us/visualstudio/code-quality/install-net-analyzers?view=vs-2022).  It's included and enabled by default in .NET 5+.

**Comment by nick-beer** [comment]
I'd take this opportunity to add a [`global.json` file](https://docs.microsoft.com/en-us/dotnet/core/tools/global-json) to the repo

**Review by nick-beer — APPROVED** [review]

**Review by jryckman — APPROVED** [review]

**Comment by jryckman** [comment]
Nick has a few comments that still need addressing and I'll put this in once they're addressed.

File: Directory.Build.props (hunk 4 above)
Comment (yaqian256) [reply Directory.Build.props]: removed.

**Comment by yaqian256** [comment]
> I'd take this opportunity to add a [`global.json` file](https://docs.microsoft.com/en-us/dotnet/core/tools/global-json) to the repo

added.  I used the same global.json as in the ASW repo.

File: src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs [hunk 5]
```diff
@@ -67,7 +67,7 @@ private void AnalyzeUsingStatement(SyntaxNodeAnalysisContext context)
             var usingStatementSyntax = (UsingStatementSyntax)context.Node;
             var containingMethodSyntax = usingStatementSyntax.Ancestors().OfType<BaseMethodDeclarationSyntax>().FirstOrDefault();
             if (containingMethodSyntax != null &&
```
Comment (pakdev) [inline src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs]: Note: sucks we can't enforce our [own conventions](https://github.com/ni/csharp-styleguide#f35-%EF%B8%8F-do-apply-consistent-formatting-on-split-statements) w/o RCS1029

File: src/NationalInstruments.Analyzers/Style/FieldsCamelCasedWithUnderscoreCodeFixProvider.cs [hunk 6]
```diff
@@ -25,14 +25,14 @@ namespace NationalInstruments.Analyzers.Style
     [Shared]
     public sealed class FieldsCamelCasedWithUnderscoreCodeFixProvider : CodeFixProvider
     {
-        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(FieldsCamelCasedWithUnderscoreAnalyzer.DiagnosticId);
+        public override sealed ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(FieldsCamelCasedWithUnderscoreAnalyzer.DiagnosticId);
```
Comment (pakdev) [inline src/NationalInstruments.Analyzers/Style/FieldsCamelCasedWithUnderscoreCodeFixProvider.cs]: Curious why this change occurred - did some analyzer flag it?

**Review by pakdev — APPROVED** [review]

File: src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs (hunk 5 above)
Comment (jryckman) [reply src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs]: I miss that. Like I said, I am positive we had it enabled in the past!

File: global.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
+{
+  "sdk": {
+    "version": "6.0.301"
```
Comment (nick-beer) [inline global.json]: I'd update this to specify `"rollForward": "patch"`, and change the version to `6.3.00` - similar to [this](https://github.com/dotnet/roslyn-analyzers/blob/58db3ef1529ce44aef5bd6324deab15f7581c27c/global.json#L17).  This will allow us to keep up to date on the latest security patches without needing to constantly modify this file.

File: src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs (hunk 5 above)
Comment (jryckman) [reply src/NationalInstruments.Analyzers/Correctness/AwaitInReadLockOrTransactionAnalyzer.cs]: I pushed a change to fix the formatting of this

File: global.json (hunk 7 above)
Comment (jryckman) [reply global.json]: Thanks for the feedback, Nick. @yaqian256 let me know when you push updates to address Nick's feedback and Pete's comments and I'll work on getting this in.

Comment (yaqian256) [reply global.json]: Updated as suggested. Thanks!

File: src/NationalInstruments.Analyzers/Style/FieldsCamelCasedWithUnderscoreCodeFixProvider.cs (hunk 6 above)
Comment (yaqian256) [reply src/NationalInstruments.Analyzers/Style/FieldsCamelCasedWithUnderscoreCodeFixProvider.cs]: I think it was done by the CodeMaid extension. But looks like "sealed override" is more popular, at least in ASW (see below), so I discard the change.

```
λ git grep "override sealed" *.cs  | wc
     13      94    1634

λ git grep "sealed override" *.cs  | wc
    346    2549   47264
```

File: src/AnalyzerConfiguration/NI.CSharp.Analyzers.targets [hunk 8]
```diff
@@ -36,17 +36,12 @@
 
   <ItemGroup Condition="'$(NI_EnableAnalyzers)' == 'True'">
     <!-- Microsoft's SA analyzers -->
-    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.dll"/>
-    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll"/>
+    <Analyzer Include="$(PkgStyleCop_Analyzers_Unstable)\analyzers\dotnet\cs\StyleCop.Analyzers.dll"/>
+    <Analyzer Include="$(PkgStyleCop_Analyzers_Unstable)\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll"/>
 
     <!-- Microsoft's CA analyzers -->
-    <Analyzer Include="$(PkgMicrosoft_CodeQuality_Analyzers)\analyzers\dotnet\cs\Humanizer.dll" />
```
Comment (yaqian256) [inline src/AnalyzerConfiguration/NI.CSharp.Analyzers.targets:43]: NOTE: these are the old packages we do not use anymore.

File: src/AnalyzerConfiguration/NI.CSharp.Analyzers.targets [hunk 9]
```diff
@@ -36,17 +36,12 @@
 
   <ItemGroup Condition="'$(NI_EnableAnalyzers)' == 'True'">
     <!-- Microsoft's SA analyzers -->
-    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.dll"/>
-    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll"/>
+    <Analyzer Include="$(PkgStyleCop_Analyzers_Unstable)\analyzers\dotnet\cs\StyleCop.Analyzers.dll"/>
```
Comment (yaqian256) [inline src/AnalyzerConfiguration/NI.CSharp.Analyzers.targets:39]: NOTE: we're depending on the beta version of StyleCop to get its latest features.  

The beta version of StyleCop.Analyzers (1.2.0-beta.435) is a thin wrapper of StyleCop.Analyzers.Unstable (1.2.0.435) , 
it does not have the `analyzers` folder. We need to get the analyzers from StyleCop.Analyzers.Unstable .
